### PR TITLE
Make noarch Python

### PR DESCRIPTION
**Pull request by SylvainCorlay**
Making `jupyter_kernel_test` noarch to save build time.

I don't know if there are blockers for packages installed with flit to be marked as noarch, so leaving it to @takluyver to merge.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -13,18 +13,16 @@ source:
...
   script: flit install --deps=none
-  skip: True  # [py2k]
 
 requirements:
   build:
-    - python
+    - python >=3
     - flit
-    - pytoml  # flit should depend on pytoml
-    - pip  # flit should depend on pip
```
Comment (SylvainCorlay) [inline recipe/meta.yaml:25]: removed `pip` and `pytoml` from the build-time dependencies because they are now properly listed as dependencies of flit.

**Review by ocefpaf — APPROVED** [review]

**Comment by takluyver** [comment]
I don't know of any issues with using flit to make noarch conda packages, but I haven't really been paying attention to how conda noarch works. Does anyone want to do anything to check, or shall we merge it and cross our fingers?

**Comment by SylvainCorlay** [comment]
Will try out locally on OSX and Linux tomorrow...

**Comment by takluyver** [comment]
Thanks

**Comment by SylvainCorlay** [comment]
Unfortunately it does not appear to work. The resulting noarch package does not function. `import jupyter_kernel_test` results in

```
In [1]: import jupyter_kernel_test
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
<ipython-input-1-1a609429c365> in <module>()
----> 1 import jupyter_kernel_test

~/dev/jupyter/jupyter_kernel_test/jupyter_kernel_test/__init__.py in <module>()
      6 from unittest import TestCase, SkipTest
      7 
----> 8 from jupyter_kernel_mgmt.discovery import KernelFinder
      9 from jupyter_kernel_mgmt.client import BlockingKernelClient, ErrorInKernel
     10 from tornado import gen

ModuleNotFoundError: No module named 'jupyter_kernel_mgmt'
```

Adding a commit exposing the failure.

**Comment by SylvainCorlay** [comment]
Actually, there is already a section about this:

```
test:
  imports:
    - jupyter_kernel_test
```

which succeeds on linux, and when building the package locally. It just happens that the actual resulting package is not usable.

**Comment by takluyver** [comment]
That looks like you've somehow managed to package master instead of a release. The line giving you the error is not in any release; it was only merged yesterday:

https://github.com/jupyter/jupyter_kernel_test/pull/41/files

**Comment by SylvainCorlay** [comment]
I rebuilt it and checked again, and could not reproduce. I was probably testing in an environment with some dev artifacts.

**Pull request merged**